bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java

```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
         Object value = jsonMap.get(key);
         return ObjectUtils.toString(value);
     }
-
+    
     private Relationship createRelationship(Person currentPerson, Map<String, Object> relationshipJson) {
-        Relationship relationship = new Relationship(currentPerson,
-                getPerson((Map<String, Object>) relationshipJson.get("personB")),
-                getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        Relationship relationship;
+        String direction = (String) relationshipJson.get("direction");
+        
+        if (direction.equals("aToB")) {
```

Comment:
I would rather use a `Constant` or better, an `Enum` to store those keywords.
And rename to `A_to_B` and `B_to_A`.